### PR TITLE
fix(kubeletConnection): leverage token refresh

### PR DESCRIPTION
**Pull request by paologallinaharbur**
Fix https://github.com/newrelic/nri-kubernetes/issues/511

Changing 
 - setting `NewBearerAuthWithRefreshRoundTripper` instead of `NewBearerAuthRoundTripper` to fix kubelet
and 
 - setting `transportConfig.BearerTokenFile` instead of  `transportConfig.BearerToken` to fix controlPlane
 - using `require` to avoid null pointer exception if creation of some resources fails and also if they are the only condition tested in a test-case (basically to remove an import)

### CODE REVIEW

**Comment by paologallinaharbur** [comment]
The security vulnerability has been fixed already in the main of client-go we should wait for a release https://github.com/kubernetes/client-go/blob/3b969f96803febcac81f3a4272a9914270fbb3a3/go.mod#L39

File: src/controlplane/client/authenticator/authenticator_test.go
Comment (marcsanmi) [reply src/controlplane/client/authenticator/authenticator_test.go:88]: Yes I prefer `assert` over `testify` when possible just to detect as many failures in a single run.  I see the benefit of mixing both but then maybe I would change to require for the authenticator constructors? I mean, it does make sense to keep running the test if creating the authenticator fails? Anyway you can omit this, as it's a nit.

> In this case the package are different but the dependency it the same github.com/stretchr/testify

I meant extra package not dependency, sorry.

Comment (paologallinaharbur) [reply src/controlplane/client/authenticator/authenticator_test.go:88]: Yes I can change for all of them at least in this package, you are right. At first I thought you meant the whole repo to remove the dependency and it was a bit too out of scope!

Comment (paologallinaharbur) [reply src/controlplane/client/authenticator/authenticator_test.go:87]: Fixed in [6fadacf](https://github.com/newrelic/nri-kubernetes/pull/512/commits/6fadacf22ac97044977999f679a63f58b52f0fa3). Thanks for pointing it out!

**Review by marcsanmi — APPROVED** [review]
LGTM (after the security update is addressed).

I cannot see any flaws, passing and empty bearer token to the `NewBearerAuthWithRefreshRoundTripper` func shouldn't have any unexpected behaviour as it's automatically recreated from the token file.

**Review by sigilioso — APPROVED** [review]
👍 LGTM
Let's wait for `go-restful` to be released

**Pull request merged**
